internal/cli/command/secret/create.go

```diff
@@ -0,0 +1,411 @@
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"gopkg.in/AlecAivazis/survey.v1"
+)
+
+const (
+	TypeGeneric = "generic"
+	TypeAmazon  = "amazon"
+	TypeAzure   = "azure"
+	TypeAlibaba = "alibaba"
+	TypeGoogle  = "google"
+	TypeOracle  = "oracle"
+)
+
+// createSecretOptions contains create secret flags for `banzai create secret` command
+type createSecretOptions struct {
+	file       string
+	secretName string
+	secretType string
+	tags       []string
+	validate   string
+}
+
+// secretFieldQuestion contains all necessary field for a secret question (any type except generic)
+type secretFieldQuestion struct {
+	input     *survey.Input
+	name      string
+	output    string
+	validator survey.Validator
+}
+
+// NewCreateCommand returns a cobra command for `banzai create create` command
+func NewCreateCommand(banzaiCli cli.Cli) *cobra.Command {
+	options := createSecretOptions{}
+
+	cmd := &cobra.Command{
+		Example: `
+	Create secret
+	---
+	$ banzai secret create
+	? Secret name mysecretname
+	? Choose secret type: password
+	? Set 'username' field: myusername
+	? Set 'password' field: mypassword
+	? Do you want to add tag(s) to this secret? Yes
+	? Tag: tag1
+	? Tag: tag2
+	? Tag: skip
+
+	Create secret with flags
+	---
+	$ banzai secret create --name mysecretname --type password --tags=cli
+	? Set 'username' field: myusername
+	? Set 'password' field: mypassword
+		`,
+		Use:     "create",
+		Aliases: []string{"c"},
+		Short:   "Create secret",
+		Long:    "Create secret based on json stdin or interactive session",
+		SilenceUsage: true,
+		Args:    cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runCreateSecret(banzaiCli, options)
+		},
+	}
+
+	flags := cmd.Flags()
+
+	flags.StringVarP(&options.file, "file", "f", "", "Secret creation descriptor file")
+	flags.StringVarP(&options.secretName, "name", "n", "", "Name of the secret")
+	flags.StringVarP(&options.secretType, "type", "t", "", "Type of the secret")
+	flags.StringArrayVarP(&options.tags, "tags", "", []string{}, "Tags to add to the secret")
+	flags.StringVarP(&options.validate, "validate", "v", "", "Secret validation (true|false)")
+
+	return cmd
+}
+
+// runCreateSecret starts to get secret properties from the user via file or survey
+func runCreateSecret(banzaiCli cli.Cli, options createSecretOptions) error {
+	out := &pipeline.CreateSecretRequest{}
+
+	if err := getCreateSecretRequest(banzaiCli, options, out); err != nil {
+		return err
+	}
+
+	log.Debugf("create secret request: %#v", out)
+
+	orgID := input.GetOrganization(banzaiCli)
+	response, _, err := banzaiCli.Client().SecretsApi.AddSecrets(
+		context.Background(),
+		orgID,
+		*out,
+		&pipeline.AddSecretsOpts{
+			Validate: getValidationFlag(options.validate),
+		},
+	)
+	if err != nil {
+		cli.LogAPIError("create secret", err, out)
+		return emperror.Wrap(err, "failed to create secret")
+	}
+
+	log.Infof("Secret (%s) created with ID: %s", out.Name, response.Id)
```

Comment:
We should print out the result using `format.SecretsWrite` as that can print out in either JSON, Yaml or Table format.